app/lib/cluster_viz_service.rb

```diff
@@ -275,4 +275,26 @@ def self.raw_matrix_for_cluster_cells(study, cluster)
 
     raw_matrix
   end
+
+  # find what annotation values are represented for this combination of cluster & annotation
+  # primarily important as a validation for differential expression, as DE cannot be run on single-value annotations,
+  # or on labels that have only one cell
+  #
+  # * *params*
+  #   - +cluster+    (ClusterGroup) => Clustering object being used as control cell list
+  #   - +annotation_name+  (String) => Name of requested annotation
+  #   - +annotation_type+  (String) => Type of requested annotation (should be 'group')
+  #   - +annotation_scope+ (String) => Scope of requested annotation ('study' or 'cluster')
+  def self.cells_by_annotation_label(cluster, annotation_name, annotation_type, annotation_scope)
+    study = cluster.study
+    cells = cluster.concatenate_data_arrays('text', 'cells')
+    annotation = { name: annotation_name, scope: annotation_scope, type: annotation_type }
```

Comment:
Blocker: We've discussed the extraneous `annotation_type` parameter for DE a few times, and typically left it as simply doing what Ingest Pipeline expects, but that strikes me as not a good enough reason to thread this extraneous parameter through all our DE code.

In other words, the fact that the DE pipeline expects an `annotation_type` parameter that only ever takes one value does not entail that all upstream code should also have that parameter.  Required parameters that take only 1 value shouldn't be parameters.  Wiring them through our interfaces is poor encapsulation -- noise that decreases readability and maintainability.

So let's omit `annotation_type` parameters from signatures in Rails DE code.

```suggestion
  def self.cells_by_annotation_label(cluster, annotation_name, annotation_scope)
    study = cluster.study
    cells = cluster.concatenate_data_arrays('text', 'cells')
    annotation = { name: annotation_name, scope: annotation_scope, type: 'group' }
```

Reply:
I won't argue the point - the parameter is technically unnecessary, and was in place solely to match the interface for DE via ingest - but it is out of scope for this PR.  The changes are more substantive than just these two edits - there are quite a few more that would need to be made, especially considering tests.  I'm happy to ticket that work and do it elsewhere, but it's not part of the scope for this PR, and shouldn't hold it up.

Reply:
@eweitz https://broadworkbench.atlassian.net/browse/SCP-4454

Reply:
Perfect, another ticket this sprint makes sense!